s6/postfix/run

```diff
@@ -91,17 +91,37 @@ if [ "${RELAYHOST_AUTH}" == "yes" ]; then
     postconf -e smtp_sasl_mechanism_filter="PLAIN LOGIN"
 fi
 
+if [ "${SENDER_DEPENDENT_RELAYHOST_AUTH}" == "yes" ]; then
+    echo "postfix >> Enabling sender dependent relayhost authentication"
+    touch /etc/postfix/relayhost_map
+    touch /etc/postfix/sasl-passwords
+    postconf -# relayhost
+    postconf -e smtp_sasl_auth_enable=yes
+    postconf -e smtp_sasl_security_options=noanonymous
+    postconf -e smtp_sasl_password_maps=hash:/etc/postfix/sasl-passwords
+    postconf -e smtp_sender_dependent_authentication=yes
+    postconf -e sender_dependent_relayhost_maps=hash:/etc/postfix/relayhost-map
+    # require encryption with relayhost auth
+    postconf -e smtp_use_tls=yes
+    postconf -e smtp_tls_security_level=encrypt
+    postconf -e smtp_sasl_mechanism_filter="PLAIN LOGIN"
+fi
+
 if [ -n "${RELAYHOST_PASSWORDMAP}" ]; then
     echo "postfix >> Generating relayhost password map"
     truncate --size 0 /etc/postfix/sasl-passwords
     IFS=',' read -r -a PASSWORDMAP <<< "${RELAYHOST_PASSWORDMAP}"
     for P in "${PASSWORDMAP[@]}"; do
         IFS=':' read -ra MAP <<< "$P"
-        if [[ "${#MAP[@]}" -eq 3 ]]; then
+        if [[ "${#MAP[@]}" -eq 2 ]]; then
+          HOST=${MAP[0]}
+          USER=${MAP[0]}
+          PASS=${MAP[1]}
+        elif [[ "${#MAP[@]}" -eq 3 ]]; then
           HOST=${MAP[0]}
           USER=${MAP[1]}
           PASS=${MAP[2]}
-        elif [[ "${#MAP[@]}" -eq 4 ]]; then
+        elif [[ "${#MP[@]}" -eq 4 ]]; then
```

Comment:
Is this a typo? (I haven't actually tested this yet)

Reply:
Yes. This is typo. I've fixed it in following commit.